qutip/core/data/dispatch.pyx

```diff
@@ -552,8 +552,9 @@ cdef class Dispatcher:
         callable object which requires that the dispatched arguments match
         those specified in `types`.
         """
-        if isinstance(types, type):
+        if isinstance(types, (type, str)):
             types = (types,)
```

Comment:
Similar comment to before about having a single source of truth - this should be just `if not isinstance(types, tuple)` now to allow `to.parse` to be the only place that defines what's acceptable (I think it's fine to assert everywhere that `tuple` will _never_ be an acceptable type).